Include vectors.json in sdist

`vectors.json` is needed for users to run all the tests.

This is useful for Linux distroes that redistribute packages built from sdist that want to run the tests as part of the packaging.

```
❯ tar xvzf mnemonic-0.20.tar.gz
mnemonic-0.20/
mnemonic-0.20/AUTHORS
mnemonic-0.20/CHANGELOG.rst
mnemonic-0.20/LICENSE
mnemonic-0.20/MANIFEST.in
mnemonic-0.20/PKG-INFO
mnemonic-0.20/README.rst
mnemonic-0.20/setup.cfg
mnemonic-0.20/setup.py
mnemonic-0.20/src/
mnemonic-0.20/src/mnemonic/
mnemonic-0.20/src/mnemonic/__init__.py
mnemonic-0.20/src/mnemonic/mnemonic.py
mnemonic-0.20/src/mnemonic/py.typed
mnemonic-0.20/src/mnemonic/wordlist/
mnemonic-0.20/src/mnemonic/wordlist/chinese_simplified.txt
mnemonic-0.20/src/mnemonic/wordlist/chinese_traditional.txt
mnemonic-0.20/src/mnemonic/wordlist/english.txt
mnemonic-0.20/src/mnemonic/wordlist/french.txt
mnemonic-0.20/src/mnemonic/wordlist/italian.txt
mnemonic-0.20/src/mnemonic/wordlist/japanese.txt
mnemonic-0.20/src/mnemonic/wordlist/korean.txt
mnemonic-0.20/src/mnemonic/wordlist/spanish.txt
mnemonic-0.20/src/mnemonic.egg-info/
mnemonic-0.20/src/mnemonic.egg-info/PKG-INFO
mnemonic-0.20/src/mnemonic.egg-info/SOURCES.txt
mnemonic-0.20/src/mnemonic.egg-info/dependency_links.txt
mnemonic-0.20/src/mnemonic.egg-info/not-zip-safe
mnemonic-0.20/src/mnemonic.egg-info/top_level.txt
mnemonic-0.20/tests/
mnemonic-0.20/tests/test_mnemonic.py
mnemonic-0.20/tools/
mnemonic-0.20/tools/generate_vectors.py
mnemonic-0.20/tox.ini

jonny in python-mnemonic on  rawhide [$?]
❯ cd mnemonic-0.20/

jonny in python-mnemonic/mnemonic-0.20 on  rawhide [$?] via 🐍 v3.11.4
❯ pytest
===================================== test session starts =====================================
platform linux -- Python 3.11.4, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/jonny/fedora-scm/python-mnemonic/mnemonic-0.20
plugins: forked-1.4.0, cov-4.0.0, xdist-3.1.0
collected 7 items

tests/test_mnemonic.py ......F                                                          [100%]

========================================== FAILURES ===========================================
__________________________________ MnemonicTest.test_vectors __________________________________

self = <test_mnemonic.MnemonicTest testMethod=test_vectors>

    def test_vectors(self) -> None:
>       with open("vectors.json", "r") as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'vectors.json'

tests/test_mnemonic.py:45: FileNotFoundError
=================================== short test summary info ===================================
FAILED tests/test_mnemonic.py::MnemonicTest::test_vectors - FileNotFoundError: [Errno 2] No such file or directory: 'vectors.json'
================================= 1 failed, 6 passed in 0.27s =================================
```

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,6 +1,7 @@
 include tools/*
 graft src
 graft tests
+include vectors.json
 
 include AUTHORS LICENSE README.rst CHANGELOG.rst
 include tox.ini

